### PR TITLE
MH-12771 Document fields of External API 1.0.0

### PR DESCRIPTION
**Pull request by staubesv**
This PR attempts to improve the precision of the External API specification.

The main focus is on including all fields of all objects in the specification (rather than just in the examples).

Note that this PR should not change the External API 1.0.0 - it rather tries (based on the current state of branch develop) to describe it in more details. In case the specification added here deviates from the current implementation, that would be a bug in the specification.

Please also note that the specification also has more room for precision: So this is an attempt to improve the specification, not to make it perfect.

### CODE REVIEW

File: docs/guides/developer/docs/api/base-api.md [hunk 1]
```diff
@@ -6,14 +6,23 @@ In order to assess key characteristics of the API and to test general connectivi
...
+`version `| [`string`](types.md#basic) | Default version of the External API
+`url`     | [`string`](types.md#basic) | Base URL clients shall use to communicate with the External API
+
+
+__Example__
+
 ```
 {
-  "url": "https:\/\/api.opencast.org",
+  "url": "https:\/\/api.opencast.org\/api",
```
Comment (staubesv) [inline docs/guides/developer/docs/api/base-api.md:25]: Just double-checked this: The default configuration does indeed return this value.

**Comment by gregorydlogan** [comment]
@staubesv: Is this a WIP PR?  Or are all the currently planned changes present?  I'm happy to review, but I'd really rather look at this once rather than have to keep coming back :)

**Comment by staubesv** [comment]
@gregorydlogan This PR is ready for review. It does document the current state of the External API 1.0.0 based on the current state of the branch develop. Happy to have you as reviewer!

**Comment by gregorydlogan** [comment]
As a heads up, this and #153 conflict with eachother.  I'm merging them locally, so I'll probably just push the fix for both.

**Comment by staubesv** [comment]
@gregorydlogan Don't worry about merge conflicts that will be caused by #153. I'll adapt this PR as soon as #153 is merged.

**Comment by staubesv** [comment]
@gregorydlogan Thanks for reviewing #153. I've resolved the merge conflicts that caused in this PR which is now ready for review.

**Comment by staubesv** [comment]
Hi @gregorydlogan I'm closing this in favor of PR #173 (the same but against 5.x)

**Pull request closed without merging**